### ISSUE TITLE
Hello! This is a temporary test" appears on localhost:8080 if you "$ twistd web"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2271]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)| @jerub reported|
|-|-|
|Trac ID|trac#2271|
|Type|enhancement|
|Created|2006-12-06 05:43:07Z|
|Branch|https://github.com/twisted/twisted/tree/twistd-web-2271-2|

Step to reproduce.
```
$ twistd web
$ curl http://localhost:8080/

            <html>
            <head><title>Temporary Test</title><head>
            <body>

            Hello!  This is a temporary test until a more sophisticated form
            demo can be put back in using more up-to-date Twisted APIs.

            </body>
            </html>
```

This should probably be replaced by something more appropriate.

Further, 'twistd web' with no args should probably give some kind of feedback. Perhaps "hTTP server started on 8080" or something equally sane.

<details><summary>Searchable metadata</summary>

```
trac-id__2271 2271
type__enhancement enhancement
reporter__jerub jerub
priority__low low
milestone__ 
branch__branches_twistd_web_2271_2 branches/twistd-web-2271-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1165383787000000 1165383787000000
changetime__1297986529000000 1297986529000000
version__None None
owner__exarkun exarkun
cc__thijs cc__mwh
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [27108]) Branching to 'twistd-web-2271'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [27109]) Removing temporary test statement, refs [#2271](https://github.com/twisted/twisted/issues/2271)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_6"></a>|@mwhudson set owner to @thijstriemstra|
|-|-|

Um, print is a little unsophisticated.  It appears during unit tests, for example.  It also is a bit misleading if the server fails to start up for some reason like the port already being bound.  Also, there's no test :-P

I'm not sure what would be better though, it seems like it would be a bit involved to only display something once the server is actually set up.  I'd be interested in having a conversation about whether this change is better or not than the status quo :)  Personally, I'm not going to approve the ticket until that conversation has happened.

The change to the resource itself seem fine.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_7"></a>|@mwhudson commented|
|-|-|

I forgot to say thanks on this one too :)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [30780]) Branching to 'twistd-web-2271-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

[#11550](https://github.com/twisted/twisted/issues/11550) or a similar ticket seems like a better place (and approach) for providing feedback about things starting (or not).  We need to wait for daemonization and service startup before we'll have any idea if anything worked.

I removed the print from `twisted/web/tap.py`, added a news fragment, and fixed a pyflake in `demo.py`.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_11"></a>|@mwhudson set owner to @exarkun|
|-|-|

The news fragment is 0 bytes long!  Dunno what happened there, but it clearly needs fixing.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_12"></a>|@mwhudson commented|
|-|-|

Now I've read ReviewProcess again, please land :-)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set status to `closed`|
|-|-|

(In [30791]) Merge twistd-web-2271-2

Author: thijs, exarkun
Reviewer: mwh
Fixes: [#2271](https://github.com/twisted/twisted/issues/2271)

Replace the static text served up by the default `twistd web` page to be
less optimistic about the future - it is a test page, anything that may
happen later is a happy bonus.
